src/plugins/data_source/config.ts

```diff
@@ -5,6 +5,8 @@
 
 import { schema, TypeOf } from '@osd/config-schema';
 import { fileAppenderSchema } from './audit_config';
+// eslint-disable-next-line @osd/eslint/no-restricted-paths
```

Comment:
What's the reason for overriding and using restricted paths?

Reply:
Using restricted path is because the interface `DataSourcePluginConfigType` is also referenced by the public module, so eslint throws `Server modules cannot be imported into client modules or shared modules`, but I see the same issue in audit_config.ts: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source/audit_config.ts#L10, so maybe it's OK.

Reply:
Can you take a look at how others have done this in the past? Ref: https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch-Dashboards+SharedGlobalConfig&type=code

These eslint rules are a really valuable mechanism used by the platform to codify supported usages/patterns, we want to be really intentional about when and why we suppress these.

Reply:
Moving `SharedGlobalConfig` to common will change many files referencing it, I've changed it by defining the timeout fields directly, so `SharedGlobalConfig` is not needed to import, please help to take a look, thanks!